docs/guides/draftjs-blocks-entityrange-entitmap.md

```diff
@@ -0,0 +1,196 @@
+
+### DraftJS block, entityRanges and entityMap
+
+A quick side note on how the DraftJS block, entityRanges and entityMap works, in the context of the TranscriptEditor component. For the [adapters](./adapters.md) guide.
+
+
+#### Data Block
+
+TL;DR: a block is a representation of a paragraph (as an Immutable Record) in draftJs and you can have some custom data associated to it.
+
+But see the docs notes on [draftjs basics](https://github.com/bbc/react-transcript-editor/blob/master/docs/notes/draftjs/2018-10-01-draftjs-1-basics.md) to better understand the role of content block within the editor. As well as the draftJs official docs.
+
+Here's an example of a block, you can see it can contain some custom data, eg speaker name, list of words, and start time (which would be the start time of the first word).
+
+```js
+[
+  {
+    "text": "There is a day.", // text 
+    "type": "paragraph", // type of block 
+    "data": { //optional custom data
+      "speaker": "TBC 0",
+      "words": [
+       ...
+      ],
+      "start": 13.02
+    },
+    "entityRanges": [ // <-- entity ranges
+   ...
+    ]
+  },
+  ...
+```
+
+It also contains a list of `entityRanges`.
+
+### Entity Ranges
+
+`entityRanges` are part of individual blocks. 
+
+<!-- See the docs notes on [draftjs entity ranges](https://github.com/bbc/react-transcript-editor/blob/master/docs/notes/draftjs/2018-10-02-drafjs-2-entity-range.md) -->
+
+From draftJs docs on [entity](https://draftjs.org/docs/advanced-topics-entities) 
+
+> the Entity system, which Draft uses for annotating ranges of text with metadata. Entities introduce levels of richness beyond styled text. Links, mentions, and embedded content can all be implemented using entities.
+
+This is what we use to identify the words, from a list of characters, and associate data to it, such as start and end time information. 
+
+It sets the foundations for features such as click on a word can jump the player play-head to the corresponding time for that word.
+
+Here's an example of `entityRanges` in the context of a data block.
+
+Required fields are the `offset`, and `length`, which are used to identify the entity within the characters of the `text` attribute of the block.
+
+This, combined with the `entityMap` has the advantage that if you type or delete some text before a certain entity, draftJs will do the ground work of adjusting the offsets and keeping these info in sync.
+
+```js
+[
```

Comment:
I think you're potentially making this confusing with the custom fields etc..?

Reply:
yeah, that's a fair point ... not sure